src/game.cpp

```diff
@@ -3552,6 +3552,16 @@ void Game::processClientEvents(CameraOrientation *cam, float *damage_flash)
 			bool enable = event.override_day_night_ratio.do_override;
 			u32 value = event.override_day_night_ratio.ratio_f * 1000;
 			client->getEnv().setDayNightRatioOverride(enable, value);
+		} else if (event.type == CE_SET_CLOUDS) {
```

Comment:
I've just tried, and it's a scoping nightmare. Currently, each "case" is its own block and may freely allocate local variables; the name "e" is very popular, for instance. After conversion to a switch/case structure, the compiler started complaining about all the variable instantiation I was skipping.

I can try placing each case statement in its own anonymous block, but that would probably violate a lot of coding guidelines and look unnecessary unless I add a lot of comments.